### PR TITLE
[WIP] [Enhancement] cache iceberg db & table

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/10032

## Problem Summary(Required) ：

The main purpose of this PR is to fix tpcds-1g query85.

We have a heuristic join reorder policy introduced in this PR: https://github.com/StarRocks/starrocks/pull/9031

But this heuristic policy is based on a assumption: same table has the same table id. But for iceberg table, that assumption is not hold. So we might have bad plan. 

To fix this problem, we have to make sure same tables have the same table id by cahcing.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 19 / 24 (79.17%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergMetadata.java|19|24|79.17%|[55, 56, 105, 119, 130]|

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -52,39 +63,71 @@ public IcebergMetadata(Map<String, String> properties) {
...
     public Table getTable(String dbName, String tblName) {
         try {
-            org.apache.iceberg.Table icebergTable
-                    = icebergCatalog.loadTable(IcebergUtil.getIcebergTableIdentifier(dbName, tblName));
-            if (IcebergCatalogType.fromString(catalogType).equals(IcebergCatalogType.CUSTOM_CATALOG)) {
-                return IcebergUtil.convertCustomCatalogToSRTable(icebergTable, catalogImpl, dbName, tblName, customProperties);
-            }
-            return IcebergUtil.convertHiveCatalogToSRTable(icebergTable, metastoreURI, dbName, tblName);
-        } catch (DdlException e) {
+            HiveTableName key = new HiveTableName(dbName, tblName);
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:128]: IMO, If we cache the IcebergTable, it seems that we couldn't get the newest currentSnapshot when construct scan range of iceberg table.

**Review by stephen-shelby — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -52,39 +63,71 @@ public IcebergMetadata(Map<String, String> properties) {
         } else {
             throw new RuntimeException(String.format("Property %s is missing or not supported now.", ICEBERG_CATALOG));
         }
+        tableCache = newCacheBuilder(MAX_TABLE_CACHE_SIZE).build(new CacheLoader<HiveTableName, Table>() {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:66]: nit: HiveTableName? may be we can add other name for this, of course it can be another pr

**Review by caneGuy — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by mxdzs0612 — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 2 above)
Comment (dirtysalt) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:66]: Yes, let's put cod refactor into another PR.

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
run starrocks_admit_test

**Comment by DorianZheng** [comment]
Maybe you can only cache `IcebergTableIdentifier -> connectorTableId` for simplification
